K8SPG-279 Added lables and annotations to doc (#301)
percona/k8spg-docs: docs/annotations.md

The following diagram illustrates this difference:

```mermaid
%%{ init : { "theme": "base", "themeVariables": {
  "primaryColor": "#93D0FF",
  "primaryTextColor": "#ffffff",
  "lineColor": "#ffffff",
  "background": "#1e1e1e"
} }}%%
graph TD
    A[Custom Resource] --> B[Operator]
    B --> C[Kubernetes resources]
    C --> D[Labels]
    C --> E[Annotations]
    D --> F[Selection]
    D --> G[Grouping]
    E --> H[External tools]
    E --> I[Documentation]
```

Both Labels and Annotations are assigned to the following objects managed by Percona Operator for PostgreSQL:

* Custom Resource Definitions
* Custom Resources
* Deployments
* Services
* StatefulSets
* PVCs
* Pods
* ConfigMaps and Secrets

## When to use labels and annotations

Use **Labels** when:

* The information is used for object selection
* The data is used for grouping or filtering
* The information is used by Kubernetes controllers
* The data is used for operational purposes

Use **Annotations** when:

* The information is for external tools
* The information is used for debugging
* The data is used for monitoring configuration

> ## Labels and annotations used by Percona Operator for MySQL

Reviewer: Here we reference MySQL Operator, but it should be Postgres

Author: Thanks, fixed

### Labels

| Name                        | Objects                          | Description                                      | Example values                          |
|-----------------------------|-----------------------------------|------------------------------------------------|-----------------------------------------|
| `pgv2.percona.com/version`  | CustomResourceDefinition          | Specifies the version of the Percona Operator for PostgreSQL. | {{release}} |
| `app.kubernetes.io/instance`  | Services, StatefulSets, Deployments | Identifies a specific instance of the application | cluster1 |
| `app.kubernetes.io/managed-by`| Services, StatefulSets           | Indicates the controller managing the object    | percona-postgresql-operator |

> | `app.kubernetes.io/component`| Services, StatefulSets           | Specifies the component within the application  | mysql, haproxy, router                  |

Reviewer: again, these components are not for postgres

Author: Fixed

| `postgres-operator.crunchydata.com/pgbackrest-backup` | Backup Jobs              | Indicates a resource that is for a pgBackRest backup.                     |
| `postgres-operator.crunchydata.com/pgbackrest-config` | ConfigMaps, Secrets      | Indicates a ConfigMap/Secret for pgBackRest.                     |
| `postgres-operator.crunchydata.com/pgbackrest-dedicated` | ConfigMaps               | Indicates a ConfigMap that is for a dedicated pgBackRest repo host.                 |
| `postgres-operator.crunchydata.com/pgbackrest-repo` | Deployments, Pods        | Indicates a Deployment or a Pod for a pgBackRest repo.      | The name of the repository you define in CR   |
| `postgres-operator.crunchydata.com/pgbackrest-volume` | PVCs     | Indicates a PVC for a pgBackRest repo volume.                |
| `postgres-operator.crunchydata.com/pgbackrest-cronjob` | CronJobs                 | Indicates a resource is a pgBackRest CronJob.                      |
| `postgres-operator.crunchydata.com/pgbackrest-restore` | Jobs, Pods               | Indicates a Job/Pod for a pgBackRest restore.                   |

> | `postgres-operator.crunchydata.com/pgbackrest-restore-config` | ConfigMaps, Secrets      | Indicates a a configuration resource (e.g. a ConfigMap or Secret) for pgBackRest restore.         |

Reviewer: ```suggestion
| `postgres-operator.crunchydata.com/pgbackrest-restore-config` | ConfigMaps, Secrets      | Indicates a configuration resource (e.g. a ConfigMap or Secret) for pgBackRest restore.         |
```

| `postgres-operator.crunchydata.com/pguser`        | Secrets, Users           | Identifies the PostgreSQL user an object is for/about.                     | Username |
| `postgres-operator.crunchydata.com/startup-instance` | Pods, Jobs               | Indicates the startup instance associated with a resource.  |
| `postgres-operator.crunchydata.com/cbc-pgrole`    | Secrets                  | Identifies a CBC PostgreSQL role secret.                                   |
| `postgres-operator.crunchydata.com/pgadmin`       | pgAdmin resources        | Indicates a resource for a standalone pgAdmin instance.                     |

### Annotations

| Name                                              | Objects  | Description                                   | Example Values              |
|---------------------------------------------------|----------|-----------------------------------------------|-----------------------------|
| `postgres-operator.crunchydata.com/trigger-switchover` | Custom Resource | Initiates a failover, switchover |             |

> | `postgres-operator.crunchydata.com/pgbackrest-backup`         | Custom Resource, Backup | Initiates a manual backup | timestamp                  |

Reviewer 2: not sure if we should document this, what if users use this annotation to bypass our backup object? @hors wdyt?

| `postgres-operator.crunchydata.com/pgbackrest-backup-job-completion`         | Restore, PVC | Added to restore jobs, pvcs, and VolumeSnapshots that are involved in the volume snapshot creation process. The annotation holds a RFC3339 formatted timestamp that corresponds to the completion time of the associated backup job. | timestamp                  |
| `postgres-operator.crunchydata.com/pgbackrest-hash`         | Custom Resource | Specifies the hash value associated with a repo configuration as needed to detect configuration changes that invalidate running Jobs (and therefore must be recreated) |                   |

> | `postgres-operator.crunchydata.com/pgbackrest-restore`         | Custom Resource | Initiates an in-place restore|  timestamp|

Reviewer 2: not sure if we should document this, what if users use this annotation to bypass our restore object? @hors wdyt?

> | `postgres-operator.crunchydata.com/pgbackrest-ip-version`         | Custom Resource | Indicates whether to use an IPv6 wildcard address for the pgBackRest "tls-server-address". Set the value "IPv6" to use an IPv6 addresses. If the annoutation is not present of has a value other than IPv6, it defaults to IPv4 (0.0.0.0). |  0.0.0.0 |

Reviewer: ```suggestion
| `postgres-operator.crunchydata.com/pgbackrest-ip-version`         | Custom Resource | Indicates whether to use an IPv6 wildcard address for the pgBackRest "tls-server-address". Set the value "IPv6" to use an IPv6 addresses. If the annotation is not present or has a value other than IPv6, it defaults to IPv4 (0.0.0.0). |  0.0.0.0 |
```

    ...
```

> ### Set Set labels and annotations for Services

Reviewer: ```suggestion
### Set labels and annotations for Services
```

    pgv2.percona.com/custom-patroni-version: "4"
  ...
```

## <a name="annotations-ignore"></a>Specifying labels and annotations ignored by the Operator

Sometimes various Kubernetes flavors can add their own annotations to the
objects managed by the Operator.

The Operator keeps track of all changes to its objects and can remove
annotations that it didn't create.

If there are no annotations or labels in the Custom Resource, the Operator does
nothing if a new label or an annotation is added to the object.

If there is an annotation or a label specified in the Custom Resource, the
Operator starts to manage annotations and labels. In this case it removes
unknown annotations and labels.

> A cloud provider can add own labels and annotations. Or you may have custom automation tools that add own labels or annotations and you need to keep them. To do this, you can specify which annotations and labels the Operator should ignore by listing them in the `spec.ignoreAnnotations` or

Reviewer 3: I think it is a copy-paste from PXCO. As I remember, we do not have ignoreAnnotations in PGO

Author: Removed. Didn't know we don't have it for PG
